### ISSUE TITLE
SNOW-610076: Snowflake-specific proxy configuration for PDO PHP driver

### ISSUE THREAD

**Issue by salvor-hardin** [snowflakedb/pdo_snowflake#285]
It is mentioned in this driver that:

```

PHP PDO Driver for Snowflake supports HTTP and HTTPS proxy connections using environment variables. 
To use a proxy server configure the following environment variables:

http_proxy
https_proxy
no_proxy
export http_proxy="[protocol://][user:password@]machine[:port]"
export https_proxy="[protocol://][user:password@]machine[:port]"

```

But this doesn't work at all. Those environment variables are ignored when using the driver.

Also, the libcurl.a file included in the driver is precompiled with an older version, away from the current one.
I'd managed to get v1.2.1 to work via proxy replacing your libcurl.a file with another one I've recompiled that forces the use of a proxy, but for 1.2.2 doesn't work.

It it is claimed the use of a proxy works, please specify how to make that work, because it doesn't

**Comment by cjsfj** [comment]
I'm able to connect using a proxy. For me these are set in the OS environment, so I had to configure php to read OS environment variables:

`variables_order = "EGPCS"`

Also fwiw, my environment variables are in all caps, not lowercase. Also, my proxy uses http:// as the protocol even though the proxy variable I use is HTTPS_PROXY. I guess this would depend on your setup, though.

**Comment by salvor-hardin** [comment]
> I'm able to connect using a proxy. For me these are set in the OS environment, so I had to configure php to read OS environment variables:
> 
> `variables_order = "EGPCS"`
> 
> Also fwiw, my environment variables are in all caps, not lowercase. Also, my proxy uses http:// as the protocol even though the proxy variable I use is HTTPS_PROXY. I guess this would depend on your setup, though.

Thanks cjsfj, will try these and see if it works

**Comment by cjsfj** [comment]
The issue is actually in libsnowflake... not the PDO driver. I opened this back in May :/

https://github.com/snowflakedb/libsnowflakeclient/issues/417

**Comment by sfc-gh-dszmolka** [comment]
hi - apologies for leaving this issue unattended for such a long time, we're changing that going forward.

as a workaround you can perhaps apply the envvars only to the PHP command you're running, something like this:
```
$ HTTPS_PROXY=my.pro.xy:8080 NO_PROXY=.mycompany.com php myapp.php
```

but we'll take a look and see how it can be implemented similarly to the other Snowflake drivers where you can specify proxy settings on the connection strings, besides in the envvars. Thank you for bearing with us !

**Comment by sfc-gh-dszmolka** [comment]
Looks like the individual Connection settings for specifying the proxy are available since v1.2.5 - created a PR https://github.com/snowflakedb/pdo_snowflake/pull/369 to document them

**Comment by sfc-gh-dszmolka** [comment]
Doc changes merged
